crates/bindings-csharp/Codegen.Tests/fixtures/server/snapshots/Module#MultiTable.verified.cs

```diff
@@ -0,0 +1,130 @@
+﻿//HintName: MultiTable.cs
+// <auto-generated />
+#nullable enable
+
+partial struct MultiTable : SpacetimeDB.Internal.ITable<MultiTable>
+{
+    public void ReadFields(System.IO.BinaryReader reader)
+    {
+        Name = BSATN.Name.Read(reader);
+        Foo = BSATN.Foo.Read(reader);
+        Bar = BSATN.Bar.Read(reader);
+    }
+
+    public void WriteFields(System.IO.BinaryWriter writer)
+    {
+        BSATN.Name.Write(writer, Name);
+        BSATN.Foo.Write(writer, Foo);
+        BSATN.Bar.Write(writer, Bar);
+    }
+
+    public readonly partial struct BSATN : SpacetimeDB.BSATN.IReadWrite<MultiTable>
+    {
+        internal static readonly SpacetimeDB.BSATN.String Name = new();
+        internal static readonly SpacetimeDB.BSATN.U32 Foo = new();
+        internal static readonly SpacetimeDB.BSATN.U32 Bar = new();
+
+        public MultiTable Read(System.IO.BinaryReader reader) =>
+            SpacetimeDB.BSATN.IStructuralReadWrite.Read<MultiTable>(reader);
+
+        public void Write(System.IO.BinaryWriter writer, MultiTable value)
+        {
+            value.WriteFields(writer);
+        }
+
+        public SpacetimeDB.BSATN.AlgebraicType GetAlgebraicType(
+            SpacetimeDB.BSATN.ITypeRegistrar registrar
+        ) =>
+            registrar.RegisterType<MultiTable>(_ => new SpacetimeDB.BSATN.AlgebraicType.Product(
+                new SpacetimeDB.BSATN.AggregateElement[]
+                {
+                    new(nameof(Name), Name.GetAlgebraicType(registrar)),
+                    new(nameof(Foo), Foo.GetAlgebraicType(registrar)),
+                    new(nameof(Bar), Bar.GetAlgebraicType(registrar))
+                }
+            ));
+    }
+
+    static IEnumerable<SpacetimeDB.Internal.TableDesc> SpacetimeDB.Internal.ITable<MultiTable>.MakeTableDesc(
+        SpacetimeDB.BSATN.ITypeRegistrar registrar
+    ) =>
+        [
+            new(
+                new(
+                    TableName: nameof(MultiTable),
+                    Columns:
+                    [
+                        new(nameof(Name), BSATN.Name.GetAlgebraicType(registrar)),
+                        new(nameof(Foo), BSATN.Foo.GetAlgebraicType(registrar)),
+                        new(nameof(Bar), BSATN.Bar.GetAlgebraicType(registrar))
+                    ],
```

Comment:
Nit for future work: we should be able to create and reuse this list instead of generating codegen for duplicates, as all table views have the same list of column names and types.